src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -34,9 +34,9 @@ namespace System.Net.Http.Headers
         Justification = "This is not a collection")]
     public abstract class HttpHeaders : IEnumerable<KeyValuePair<string, IEnumerable<string>>>
     {
-        private Dictionary<string, HeaderStoreItemInfo> _headerStore;
-        private Dictionary<string, HttpHeaderParser> _parserStore;
-        private HashSet<string> _invalidHeaders;
+        private Dictionary<HeaderDescriptor, HeaderStoreItemInfo> _headerStore;
+        private HttpHeaderType _allowedHeaderTypes;
+        private HttpHeaderType _treatAsCustomHeaderTypes;
```

Comment:
This is removing the dictionary and the hashset from every headers object?  That's excellent.